### DISCUSSION TITLE
Support for Combo Charts in Queries

### DISCUSSION THREAD

**Discussion by vanhaito** [RamiAwar/dataline#347] [Q&A] [answered]
Dear @RamiAwar,

I would like to request the ability to create more than one chart in a single query. For example, a query like: "Use a line chart to show the sale amount performance in 2023 and a bar chart for 2024".

If possible, it would be great if dataline could support visualizing combo charts, such as:

A combo bar and line chart.
Multiple line charts for different data segments.
This feature would add flexibility and enhance data representation capabilities within the platform. Thank you for considering this request, and I look forward to your feedback!

**Comment by RamiAwar** [comment]
Hey! You can already ask for multiple results in one query actually.

It doesn't always work great but it works!

**Reply by vanhaito** [reply]
Actually I can ask for multiple results. However, when i ask a query "Use a line chart to show the sale amount performance in 2023 and a bar chart for 2024", it only visualizes the bar chart not the combo chart. How can I visualize combo chart for questions like this?

**Reply by RamiAwar** [reply, marked as answer]
Ah now I see what you mean.

Combo charts are quite a step up in difficulty unfortunately. We do plan on supporting them eventually, but it's not as trivial as other types of charts.

If you're interested in helping us get there, basically the flow is generating ChartJS json using LLMs, but also validating that the SQL generated fits perfectly into whatever template is selected. So in case you're looking at a combo chart, one example of that is 2 line charts overlayed. This means the SQL results should be 3 rows: one is the shared X-axis values, and the other two are the different Y-axis values.

In case of a combo-chart, you're also looking at a more dynamic validation and construction. So if the user asks for a line and bars overlayed, then you need to turn that into some structured output behind the scenes, and build a very accurate validation for it, and convert it to chartjs json. The main difficulty in building something like this while still keeping it in sync with your DB is that you can't just let the LLM lead the way. There's very delicate SQL involved and it needs to be in-sync with whatever charts are generated.

I think charting in general can be made into it's own open source project cause there's so much complexity there tbh 😄 
Was one of my favorite parts, I'd love to spend more time on it, just not a big priority for our *current* users. If more people wanted it I'd prioritize it more.
